analytics/analytics/job_processor/build_timings.py

```diff
@@ -94,7 +97,13 @@ def get_package_mapping(timings: list[dict]) -> dict[str, PackageDimension]:
 
 
 def create_build_timing_facts(job_fact: JobFact, gljob: ProjectJob):
-    timings = [t for t in get_timings_json(gljob) if t.get("name") and t.get("hash")]
+    # Sometimes the timings file isn't present, even on successful jobs.
+    # If that's the case, we just skip this step.
+    try:
+        data = get_timings_json(gljob)
+        timings = [t for t in data if t.get("name") and t.get("hash")]
+    except JobArtifactFileNotFound:
+        return
```

Comment:
[Here](https://gitlab.spack.io/spack/spack/-/jobs/12503050) is an example of a successful job missing an `install_times.json` file. It appears it was due to a TLS timeout.

So there's two ways we could treat an error like this.
1. It's an error in _our_ system, and so we should be aware of it. In that case, we'd want to _not_ gracefully handle the error.
2. It's a transient error that we can never fully avoid. In that case, we should gracefully handle those errors, and not raise it to sentry.

This case unfortunately seems like a bit of both. It's a transient error in nature, although it's a known systemic issue in our cluster. I'm of the mindset that we should gracefully handle this type of error, and if we want to get more info about how often this happens, we would augment the error classifier to check the log trace for the relevant text (and check successful jobs as well I suppose).

Curious what your thoughts are @mvandenburgh. I guess the long-winded answer to your question is that this is _not_ a legitimate spack error.


In either case, one thing I definitely should do is split up the main fact generation step and build timing steps into two transactions, to account for this scenario. I'll go ahead and do that.

Reply:
Although I may stand corrected? [Here](https://gitlab.spack.io/spack/spack/-/jobs/12503054) is an example of a successful job missing an `install_times.json` file, although there's seemingly no error shown in the log trace. It only mentions something about the `after_script`, which might be where build timing stuff happens, I'm not sure. So _maybe_ this is a legitimate bug in spack, maybe not. Either way, it seems there are various scenarios where that file can be missing from successful jobs.

Reply:
generate jobs will not have an `install_times.json`, which I think we are capturing with this now. Build jobs that en up being `rebuild nothing` will also have no `install_times.json`.

If there is a build, and it did rebuild stuff, and it passed, `install_times.json` _should_ exist I believe...I am not able to think of a case it would not off the top of my head.